### ISSUE TITLE
skip_commit is only about the HEAD commit

### ISSUE THREAD

**Issue by akimd** [appveyor/ci#1961]
Hi!

We're building a large app for Windows and macOS from a common repo, because many bits are common.  However, there are files that are specific to macOS and metadata files that should not trigger AppVeyor builds.

`skip_commit.files` seems like the right tool.  However the builds do not trigger if the _last_ commit of a PR is about one of these ignored files.  Yes, this is documented (https://www.appveyor.com/docs/how-to/filtering-commits/#skip-commits-2):
> `skip_commits.files` allows skipping AppVeyor build if all of the files modified in push’s *head commit* match any of the file matching rules

however, this is hardly the most natural expectation for checking a PR.

Are there any plans to move to a range-of-commits based approach?

Cheers.

**Comment by IlyaFinkelshteyn** [comment]
@akimd yes, we track in https://github.com/appveyor/ci/issues/1923. One of ideas is to add separate setting for  range-of-commits, to preserve existing behavior. One issue with PR though is that webhook does not contain non-head commit information and it can hit performance if we need to call GitHub API for large commit range.
